server/memcached/src/main/java/org/infinispan/server/memcached/MemcachedDecoder.java

```diff
@@ -296,6 +301,26 @@ private void decodeValue(ChannelHandlerContext ctx, ByteBuf buffer, MemcachedDec
       writeResponse(ch, ret);
    }
 
+   private Object touch() {
+      final CacheEntry<byte[], byte[]> cacheEntry = cache.getCacheEntry(key);
+      if (cacheEntry == null) {
+         return createNotExistResponse();
```

Comment:
It looks like this should return "NOT_FOUND\r\n" instead.

Reply:
createNotExistResponse() actually returns a NOT_FOUND message ... I haven't named this already existing method :)

Reply:
Great! Sorry, I didn't look more closely :)